craft_platforms/charm/_build.py

```diff
@@ -33,14 +33,134 @@
 """
 
 
+def _validate_base_definition(
+    base: Optional[str],
+    build_base: Optional[str],
+    platform_name: Optional[str],
+    platform: Optional[_platforms.PlatformDict],
+) -> None:
+    """Validate that a base is defined correctly in the data used to create a build.
+
+    The rules are:
+     - a base must be defined in only one place
+     - each platform must build on and build for the same base
+
+    :raises ValueError: If the base is not defined correctly in the build data.
+    """
+    if not (platform_name or base or build_base):
+        raise ValueError("No base, build-base, or platforms are specified.")
+
+    if platform_name:
```

Comment:
I think this would be more readable as:

```suggestion
    if not platform_name:
        return
```